packages/coffeescript/plugin/compile-coffeescript.js

```diff
@@ -69,20 +72,30 @@ export class CoffeeCompiler extends CachingCompiler {
       return null;
     }
 
+    let sourceMap = JSON.parse(output.v3SourceMap);
+
     if (source.indexOf('`') !== -1) {
       // If source contains backticks, pass the coffee output through babel-compiler
-      const doubleRoastedCoffee = this.babelCompiler.processOneFileForTarget(inputFile, output.js);
-      if (doubleRoastedCoffee != null && doubleRoastedCoffee.data != null) {
+      const doubleRoastedCoffee =
+        this.babelCompiler.processOneFileForTarget(inputFile, output.js);
+
+      if (doubleRoastedCoffee != null &&
+          doubleRoastedCoffee.data != null) {
         output.js = doubleRoastedCoffee.data;
+
+        // Combine the original CoffeeScript source map with the one
+        // produced by this.babelCompiler.processOneFileForTarget.
+        const smg = new SourceMapGenerator(
+          new SourceMapConsumer(doubleRoastedCoffee.sourceMap));
+        smg.applySourceMap(new SourceMapConsumer(sourceMap));
+        sourceMap = smg.toJSON();
       }
```

Comment:
Do you have any before and after examples showing what difference this change makes? I don’t see how the Babel sourcemap would be any different from the CoffeeScript one, for the code within backticks. Is the line mapping different for breakpoints or something like that?